Update IPA Installation Techniques and Tools (by @NVISOSecurity) (#3100)
OWASP/owasp-mastg: techniques/ios/MASTG-TECH-0090.md

If you want to enable dynamic testing with Frida but don't have access to a jailbroken device, you can patch and repackage the target app to load the [Frida gadget](https://www.frida.re/docs/gadget/). This way, you can instrument the app and do everything you need to do for dynamic analysis (of course, you can't break out of the sandbox this way). However, this technique only works if the app binary isn't FairPlay-encrypted (i.e., obtained from the App Store).

On a jailbroken device, you can run `frida-server` which will take care of the injection for you, even in encrypted apps. However, on a non-jailbroken device we have to manually prepare the application. There are two approaches we can take:

- Install a debug version of the application and inject during application launch
- Repackage the application to already include the Frida Gadget

As an alternative to this automated approach, see @MASTG-TECH-0091.

> You can inject Frida into an application using @MASTG-TOOL-0039, @MASTG-TOOL-0118 or @MASTG-TOOL-0038

Reviewer: The ID MASTG-TOOL-0039 is correct for Frida, but we don't explain how to inject Frida into an app running on a non-jailbroken device. Shall we link to the Frida documentation instead?

```suggestion
You can inject Frida into an application using @MASTG-TOOL-0118 or @MASTG-TOOL-0038, or follow the [instructions from the Frida website](https://frida.re/docs/ios/#without-jailbreak).
```

Author: I've removed tool 39 and the link to the frida website. This technique is explaining how to do it on a non-jailbroken, so would be weird to link to the frida docs this way.

> You can inject Frida into an application using @MASTG-TOOL-0039, @MASTG-TOOL-0118 or @MASTG-TOOL-0038

Author: ```suggestion
You can inject Frida into an application using @MASTG-TOOL-0118 or @MASTG-TOOL-0038
```

## Frida

> After following any of the techniques of @MASTG-TECH-0057, your application will be running with the `get-task-allow` entitlement, which means it can be debugged. This means that the `frida` CLI tool can spawn the application and inject the Frida Gadget automatically, even on non-jailbroken devices.

Reviewer 2: Are you sure it's a correct technique ID? It points to `Listing Installed Apps`

> First, download the latest version of the Frida Gadget and move it to `/Users/<USER>/.cache/frida/gadget-ios.dylib`:

Reviewer: ```suggestion
First, download the latest version of the Frida Gadget and move it to `/Users/<USER>/.cache/frida/gadget-ios.dylib`. Frida is released frequently, so find the latest version available on the [Github releases page](https://github.com/frida/frida/releases) or download on the command line:
```

```bash

> wget https://github.com/frida/frida/releases/download/16.5.9/frida-gadget-16.5.9-ios-universal.dylib.gz

Reviewer: Let's tell them to always use the latest version. Today it's already 16.6.1.

```suggestion
$ wget https://github.com/frida/frida/releases/download/X.Y.Z/frida-gadget-X.Y.Z-ios-universal.dylib.gz
```

> gzip -d frida-gadget-16.5.9-ios-universal.dylib.gz

Reviewer: ```suggestion
$ gzip -d frida-gadget-X.Y.Z-ios-universal.dylib.gz
```

> mv frida-gadget-16.5.9-ios-universal.dylib /Users/MAS/.cache/frida/gadget-ios.dylib

Reviewer: ```suggestion
mv frida-gadget-X.Y.Z-ios-universal.dylib /Users/MAS/.cache/frida/gadget-ios.dylib
```
